vllm/compilation/reshapes.py

```diff
@@ -50,18 +66,48 @@ def __call__(self, graph: torch.fx.Graph):
                     # Invalid reshape args, skip
                     continue
 
-                if all(
-                        self.dims_equivalent(s, i_s)
-                        for s, i_s in zip(shape, input_shape)):
+                if self.all_dims_equivalent(shape, input_shape):
                     node.replace_all_uses_with(input)
                     graph.erase_node(node)
                     count += 1
 
+            elif is_func(node, torch.ops.aten.slice.Tensor):
+                input, dim_index, start, end = node.args[:4]
+                input_shape = input.meta["val"].shape
+                i_dim = input_shape[dim_index]
+
+                if start == 0 and self.dims_equivalent(end, i_dim):
+                    node.replace_all_uses_with(input)
+                    graph.erase_node(node)
+                    count += 1
+
+            elif is_func(node, torch.ops.aten.slice_scatter.default):
```

Comment:
Are these always the right ops to use?  e.g. is there a torch.ops.aten.slice.default or a torch.ops.aten.slice_scatter.Tensor?

Reply:
I haven't seen them so I am not sure - I just went off what I saw. The other overloads could be added easily if we ever see them in the graph